raft-tech/TANF-app
# Kibana Dashboard MVP

## Summary of Changes
- Added support for automated Kibana object upload
- Created four dashboards corresponding to the `Families` section in [this](https://www.acf.hhs.gov/sites/default/files/documents/ofa/fy2023_tanf_caseload.pdf) document.
- Updated Terraform to specify the Elastic version and allow for a long timeout because on average it takes longer than the default timeout of 15 minutes for cloud.gov/AWS to stand up the instance.
Pull request closes #2883 

Note: I did not create dashboards/visualizations for every single table in the linked document. If that is what we want, I can do so. Just didn't want to do too much.

## How to Test
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in as a Sys admin.
2. In an incognito window: Open http://localhost:3000/ and sign in as a data analyst with a different STT.
3. For Both STTs submit TANF S1 and S3 data for FY 2021 Quarter 1. For Section 1, use `ADS.E2J.FTP1.TS06` and for Section 3, use the attached file below. Note, this file is a modified version of the original: `ADS.E2J.FTP3.TS06`.
[ADS.E2J.FTP3.TS06.txt](https://github.com/raft-tech/TANF-app/files/15380312/ADS.E2J.FTP3.TS06.txt)
4. [Local only] Make sure to run `source commands.sh` and then execute the new command: `upload_kibana_objs`
5. As the sys admin user, navigate to Kibana->Stack Management->Saved Objects and verify you see have several new saved objects.
6. From there, navigate through the `Visualizations` and `Dashboards`. Make sure your time range includes the datetime that you submitted your S1/S3 files or there will be no data to see.



## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] Create necessary queries for reports
- [ ] Create dashboards mirroring: [this](https://www.acf.hhs.gov/sites/default/files/documents/ofa/fy2023_tanf_caseload.pdf) 
- [ ] Objects are automatically loaded into Kibana on deployment

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
